src/frontend/context/ManuellJournalførContext.tsx

```diff
@@ -0,0 +1,634 @@
+            );
+        }
+    };
+
+    const [tilknyttedeBehandlingIder, settTilknyttedeBehandlingIder] = React.useState<number[]>([]);
+
+    const finnDokument = (
+        ressurs: Ressurs<IDataForManuellJournalføring>,
+        dokumentInfoId: string | undefined
+    ) => {
+        return ressurs.status !== RessursStatus.SUKSESS
+            ? undefined
+            : ressurs.data.journalpost.dokumenter?.find(
+                  dokument => dokument.dokumentInfoId === dokumentInfoId
+              );
+    };
+
+    const finnValgtDokument = (ressurs: Ressurs<IDataForManuellJournalføring>) => {
+        return finnDokument(ressurs, valgtDokumentId);
+    };
+
+    const settValgtDokumentInfo = (dokumentInfo: IDokumentInfo) => {
+        oppdatertData.status === RessursStatus.SUKSESS &&
+            settOppdatertData({
+                ...oppdatertData,
+                data: {
+                    ...oppdatertData.data,
+                    journalpost: {
+                        ...oppdatertData.data.journalpost,
+                        dokumenter: oppdatertData.data.journalpost.dokumenter?.map(dokument => {
+                            return dokument.dokumentInfoId === valgtDokumentId
+                                ? dokumentInfo
+                                : dokument;
+                        }),
+                    },
+                },
+            });
+    };
+
+    const settDokumentTittel = (dokumentTittel: string) => {
+        const valgt = finnValgtDokument(oppdatertData);
+        if (!valgt) {
+            return;
+        }
+        valgt.tittel = dokumentTittel;
+        valgt.brevkode = hentBrevkode(valgt.tittel);
+        settValgtDokumentInfo(valgt);
+    };
+
+    const settLogiskeVedlegg = (logiskeVedleggNavn: Array<string>) => {
+        const valgt = finnValgtDokument(oppdatertData);
+        if (!valgt) {
+            return;
+        }
+
+        valgt.logiskeVedlegg = logiskeVedleggNavn.map(vedlegg => {
+            return {
+                tittel: vedlegg,
+                logiskVedleggId: '0', // this id is not nullable but ignored by backend so set it to whatever string
+            };
+        });
+        settValgtDokumentInfo(valgt);
+    };
+
+    const hentBrevkode = (dokumentTittel: string | undefined): string => {
+        return BrevkodeMap.get(dokumentTittel) || '';
+    };
+
+    const settFagsak = (fagsak: Ressurs<IFagsak | undefined>) => {
+        if (
+            oppdatertData.status === RessursStatus.SUKSESS &&
+            fagsak.status === RessursStatus.SUKSESS
+        ) {
+            settOppdatertData({
+                ...oppdatertData,
+                data: {
+                    ...oppdatertData.data,
+                    fagsak: fagsak.data,
+                },
+            });
+        }
+    };
+
+    const hentFagsak = async (personId: string) => {
+        return axiosRequest<IFagsak | undefined, Ressurs<IFagsak | undefined>>({
```

Comment:
We should change the API to expect the ident in the body to prevent logging this :) Talked to Geirmund about this.

Reply:
Ok